### PR TITLE
Compare branch contents without immediate merge

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Compare branch contents without immediate merge

Correct the checklist phrasing to note that we need to review changes from the master branch but we don't want to merge them without discussion and consideration of the impact of each of the differences.

https://github.com/jenkinsci/packaging/pull/332#issuecomment-1238318777o is the inspiration for this pull request.

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/1-lts-release-checklist.md [hunk 1]
```diff
@@ -66,9 +66,9 @@ This role should rotate between LTS releases
 
 - [ ] Publish changelog (one day prior to the release in case of a security update)
 
-- [ ] Assure that contents of master branch have been merged to the stable branch in the [release repository](https://github.com/jenkins-infra/release), e.g. `stable-2.277`
+- [ ] Compare the contents of the [release repository](https://github.com/jenkins-infra/release) master branch with the stable branch (e.g. `stable-2.361`).  Discuss differences in the Jenkins developers mailing list in case one or more of those differences should be backported
```
[the remark this answers is not in the document]
Comment (MarkEWaite) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:69]: Do we need a new heading that is "a day or two before release"?  This would move to that section so that there is time to discuss it before the release.

[the remark this answers is not in the document]
Comment (timja) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:69]: I don’t think it’s worth it, these branches don’t differ much and generally it’s easier to keep them as close as possible. The times we want to keep them different is incredible rare.

but failing a release because some infra change wasn’t backported is more often

[the remark this answers is not in the document]
Comment (NotMyFault) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:69]: Would it make sense to get a rid of both points? Do we actually need the content on the master branch on a stable branch outside the backporting process?

[the remark this answers is not in the document]
Comment (dduportal) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:69]: @NotMyFault yes some changes must be backported from the principal branch to the `stable` branch at least on the infrastructure point of view for the pipelines (`Jenkinsfile.d/*`), the pod templates associated to these pipelines (`PodTemplates.d/*`)

[the remark this answers is not in the document]
Comment (dduportal) [reply .github/ISSUE_TEMPLATE/1-lts-release-checklist.md:69]: I've opened this issue a few months ago: https://github.com/jenkins-infra/release/issues/250 about this topic.
TL;DR; as much as the "stable branches" model fits the core, the packaging and the core Docker image, it's harder to justify here and a "only a main branch" pattern is proposed.
